### PR TITLE
Pull Request for Issue1833: Implementing beam status controls for Main

### PR DESCRIPTION
**Pull request by helfrij**
Generalized the existing beam status class to apply to BioXASSide and BioXASMain.

### CODE REVIEW

File: source/beamline/AMEnumeratedControl.cpp [hunk 1]
```diff
@@ -109,7 +109,12 @@ bool AMEnumeratedControl::addOption(int index, const QString &optionString)
 
 		indexStringMap_.insert(index, optionString);
 
+//		updateEnumStates();
```
Comment (dretrex) [inline source/beamline/AMEnumeratedControl.cpp]: Do you want to hang onto these commented lines?

Comment (helfrij) [inline source/beamline/AMEnumeratedControl.cpp]: Actually, no. I'll take them out.

**Comment by dretrex** [comment]
Okay, otherwise, looks good.
Time Estimate: 10 minutes

File: source/beamline/AMEnumeratedControl.cpp (hunk 1 above)
Comment (helfrij) [inline source/beamline/AMEnumeratedControl.cpp]: Commented out lines have been removed.

**Pull request merged**